doc/Plugin-API.md

```diff
@@ -6,12 +6,16 @@ In the following, we focus on the mechanics of Theia plugins and Theia’s compa
 This documentation aims to support developers extending Theia’s plugin API to either enhance the extensibility of Theia via plugins and/or increase Theia’s coverage of the VS Code Extension API – and with that the number of VS Code extensions that can be used in Theia.
 
 Theia plugins, as well as VS Code extensions, can be installed and removed from a Theia installation at runtime and may extend many different capabilities of Theia, such as theming, language support, debuggers, tree views, etc., via a clearly defined API.
+A special category of "headless" plugins addresses more application-specific extension points, not intended to extend Theia's out-of-the-box capabilities.
+For the most part, the discussion in this document of the plugin API does not apply to these plugins but only to VS Code compatible plugins.
```

Comment:
Could we maybe exchange that with a more specific sentence that highlights the similarities or differences to the other plugins. It might be enough to state that that the Theia plugin API aims to be a superset VS Code extension API and is therefore more concerned with the user facing aspects of the application, whereas headless plugins are meant as a hook for application-specific API that only runs in the backend. We could even consider having a picture explaining how the Theia/VS Code plugins differ from the headless plugins if that helps.

I just feel that as such a vague statement makes a user doubt the completeness of this documentation a bit - and for no reason as you did a great job in extending the the different sections!